packages/lib/src/vis/heatmap/HeatmapMesh.tsx

```diff
@@ -174,21 +110,25 @@ function HeatmapMesh(props: Props) {
     return new DataTexture(colors, CMAP_SIZE, 1, RGBFormat, UnsignedByteType);
   }, [colorMap, invertColorMap]);
 
-  const scaleUniforms = isScaleType(scaleType)
-    ? SCALE_SHADER[scaleType].uniforms(domain)
-    : SCALE_SHADER[ScaleType.Gamma].uniforms(domain, scaleType[1] as number);
+  const [scaleType, gammaExponent] = isScaleType(visScaleType)
+    ? [visScaleType]
+    : visScaleType;
+
+  const scaledDomain = scaleDomain(domain, scaleType as ScaleType);
 
   const shader = {
     uniforms: {
       data: { value: dataTexture },
-      withAlpha: { value: alphaValues ? 1 : 0 },
-      alpha: { value: alphaTexture },
       colorMap: { value: colorMapTexture },
-      alphaMin: { value: alphaDomain?.[0] },
+      min: { value: scaledDomain[0] },
+      oneOverRange: { value: 1 / (scaledDomain[1] - scaledDomain[0]) },
+      gammaExponent: { value: gammaExponent },
```

Comment:
These three uniforms are now passed here instead of via `scaleUniforms` (now removed). `gammeExponent` just ends up being `undefined` if the scale is not `Gamma`.

Reply:
All right, why not